### PR TITLE
Video chapters

### PR DESCRIPTION
**Pull request by mr-mr**
First instance of video CHAPTERS done.
Data is populated from the google sheet

image

Mucho tweaks to be done, @akash1810 suggested a separate PR for any cosmetic and functions (e.g playing by clicking the chapters)

gif-keyboard-2426459399090262637

### CODE REVIEW

**Comment by duarte** [comment]
Nice! Is it better to keep it on a separate branch while we work on tweaking visuals and features?

File: src/css/_docs.scss [hunk 1]
```diff
@@ -342,3 +342,39 @@
...
+    font-size: 15px;
+    line-height: 17px;
+    border: 1px solid $c-grey-mid;
+    &:hover {
+      background-color: darken($c-background, 3%);
+    }
+    &:before {
+      color: $c-white;
+      counter-increment: chapters;
+      content: "Chapter " counter(chapters) ": ";
```
Comment (akash1810) [inline src/css/_docs.scss]: why do this in css rather than js?

File: src/js/lib/youtube.js [hunk 2]
```diff
@@ -21,6 +21,15 @@ export function pimpYouTubePlayer(videoId, node, height, width) {
             function playerReady() {
                 youTubePlayer.playVideo();
             }
+
+            var chapterBtns = [].slice.call(document.querySelectorAll('.docs--chapters li'));
+
+            chapterBtns.forEach( function(chapterBtn){
+              chapterBtn.onclick = function(){
+                var chapTime = parseInt(chapterBtn.getAttribute('data-sheet-attribute'));
```
Comment (akash1810) [inline src/js/lib/youtube.js]: `data-sheet-attribute` is a bit of an odd name, would `data-chapter-timestamp` be more explicit?

File: src/js/main.js [hunk 3]
```diff
@@ -14,22 +14,25 @@ export function init(el, context, config) {
...
         var hiddenDesc = builder.querySelector('.docs--standfirst-hidden');
         var showMoreBtn = builder.querySelector('.docs--standfirst-read-more');
 
         var hiddenAbout = builder.querySelector('.docs--about-wrapper');
         var showAboutBtn = builder.querySelector('.docs--sponsor-aboutfilms');
         var hideAboutBtn = builder.querySelector('.docs--about-wrapper');
 
+        var chapterButtons = builder.querySelector('.docs--chapters');
+        chapters.forEach( function(chapter){
+          chapterButtons.innerHTML += '<li class="" data-sheet-attribute="'+ chapter.chapterTimestamp +'">' + chapter.chapterTitle + '</li>';
```
Comment (akash1810) [inline src/js/main.js]: for a subsequent PR, it may be better to use `appendChild` rather than concatenating to `innerHTML`

File: src/js/main.js [hunk 4]
```diff
@@ -14,22 +14,25 @@ export function init(el, context, config) {
...
         var hiddenDesc = builder.querySelector('.docs--standfirst-hidden');
         var showMoreBtn = builder.querySelector('.docs--standfirst-read-more');
 
         var hiddenAbout = builder.querySelector('.docs--about-wrapper');
         var showAboutBtn = builder.querySelector('.docs--sponsor-aboutfilms');
         var hideAboutBtn = builder.querySelector('.docs--about-wrapper');
 
+        var chapterButtons = builder.querySelector('.docs--chapters');
+        chapters.forEach( function(chapter){
+          chapterButtons.innerHTML += '<li class="" data-sheet-timestamp="'+ chapter.chapterTimestamp +'">' + chapter.chapterTitle + '</li>';
```
Comment (akash1810) [inline src/js/main.js]: the `class=""` can be removed

**Comment by akash1810** [comment]
couple of minor comments, but LGTM 👍 to merge and refactor in future PRs.

File: src/js/lib/youtube.js [hunk 5]
```diff
@@ -21,6 +21,15 @@ export function pimpYouTubePlayer(videoId, node, height, width) {
             function playerReady() {
                 youTubePlayer.playVideo();
             }
+
+            var chapterBtns = [].slice.call(document.querySelectorAll('.docs--chapters li'));
```
Comment (akash1810) [inline src/js/lib/youtube.js]: We're going to have [two versions of the same thing soon](https://github.com/guardian/docs-interactive-template/pull/21/files#diff-691ff2b74e21c3011d7cec25c48f6146R48). We should abstract it out into a module that's shared to keep things DRY.

**Pull request merged**